README.md

```diff
@@ -1 +1,167 @@
+function processResponse1(error, data1) {
+  ...
+}
+
+callOne({...}, processResponse1)
+```
+
+or
+
+```js
+var processResponse1 = function(error, data1) {
+  ...
+}
+
+callOne({...}, processResponse1)
+```
+
+The difference is that the named function will be hoisted, meaning its definition can be anywhere in the file while the variable function's definition must precede the usage of it. Another difference between named functions and variables is that named function will have names in the stack trace (the error message for debugging) so they are better for trouble shooting and debugging..
+
+Now where did the other two request go? They are inside of the `processResponse1`:
+
+```js
+callOne({...}, processResponse1)
+
+function processResponse1(error, data1) {
+  callTwo(data1, processResponse2)
+}
+
+function processResponse2(error, data2) {
+  callThere(data2, processResponse3)
+}
+
+function processResponse3(error, data1) {
+  ...
+}
+```
+
+This code is considerably easier to read, but that's not all we can do to improve it. We can take it a step further.
+
+Note: `callOne`, `callTwo`, etc. are just pseudo code for making HTTP requests. In real-life example you would substitute them for real functions using libraries like `http`, `request` or `superagent`. For now it's not really important what those functions do as long as they are asynchronous and use callbacks.
+
+## Using Modules to Mitigate Callback Hell
+
+Imagine having to work on a file with 10 named functions as listed in the previous section. You have 5 co-workers working on the same file. Each callback/response is 50 lines of code. That's insane! Of course it's better than having them in the closures (i.e., anonymous function definitions), but we know how to use modules so let's apply them by making each callback a separate module:
+
+```js
+var processResponse1 = require('./response1.js')
+callOne({...}, processResponse1)
+```
+
+This is the `response1.js` file:
+
+```js
+var processResponse2 = require('./response2.js')
+module.exports = function processResponse1(error, data1) {
+  callTwo(data1, processResponse2)
+}
+```
+
+This is the `response2.js` file:
+
+```js
+var processResponse3 = require('./response3.js')
+module.exports = function processResponse2(error, data2) {
+  callThere(data2, processResponse3)
+}
+```
+
+This is the `response3.js` file:
+
+```js
+module.exports = function processResponse3(error, data3) {
+  ...
+}
+```
+
+
+So now your 5 co-workers will love you even more because each piece of logic will be in its own file (you can easily separate files into different folder by providing paths to the `require`). Another added benefit of modular approach is that you and your team can write unit tests for each module. Having unit tests greatly increases robustness of the system and the velocity of development, because developers can make changes without being afraid of unintentionally breaking existing functionality (if they break the tests will catch that!).
```

Comment:
This feels like a side discussion -- it's really the same as the approach taken above, only now the functions are in separate files. I'm not sure we need this section. (And also agree that at the very least we don't need to mention unit tests again.)